docs/docs/creating-a-local-plugin.md

```diff
@@ -4,24 +4,68 @@ title: Creating a Local Plugin
 
 If a plugin is only relevant to your specific use-case, or if you’re developing a plugin and want a simpler workflow, a locally defined plugin is a convenient way to create and manage your plugin code.
 
+## Project structure for a local plugin
+
 Place the code in the `plugins` folder in the root of your project like this:
 
-```text
-plugins
-└── my-own-plugin
-    └── package.json
+```
+/my-gatsby-site
+└── gatsby-config.js
+└── /src
+└── /plugins
+    └── /my-own-plugin
+        └── package.json
+```
+
+The plugin also needs to be added to your `gatsby-config.js`, because there is no auto-detection of plugins. It can be added alongside any other 3rd party Gatsby plugins already included in your config.
+
+For the plugin to be discovered when you run `gatsby develop`, the plugin's root folder name needs to match the name used in the `gatsby-config.js` (_not_ its _name_ in its `package.json` file). For example, in the above structure, the correct way to load the plugin is:
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    `gatsby-third-party-plugin`,
+    `my-own-plugin`, // highlight-line
+  ],
+}
 ```
 
-**NOTE:** You still need to add the plugin to your `gatsby-config.js`. There is no auto-detection of local plugins.
+Then your plugin can begin to hook into Gatsby through Node and SSR APIs.
 
-**NOTE:** For the plugin to be discovered, the plugin's root folder name is the value that needs to be referenced in order to load it (_not_ its _name_ in its package.json file). For example, in the above structure, the correct way to load the plugin is:
+## Developing a local plugin that is outside your project
+
+Your plugin doesn't have to be in your project in order to be tested or worked on. If you'd like to [decouple](/docs/glossary#decoupled) your plugin from your site so it can be published as its own package, or you'd like to test or develop a forked version of a community authored plugin, you can follow one of the methods described below.
+
+### Using `require.resolve` and a filepath
+
+In addition to including a plugin in the `plugins` folder, you can give the path to a plugin included in a different location on your machine and `require` it.
 
 ```javascript:title=gatsby-config.js
 module.exports = {
-  plugins: ["my-own-plugin"],
+  plugins: [
+    `gatsby-plugin-react-helmet`,
+    // highlight-start
+    {
+      // including a plugin from outside the plugins folder needs the path to it
+      resolve: require.resolve(`../path/to/gatsby-local-plugin`),
+    },
+    // highlight-end
+  ],
 }
 ```
 
+### Using `npm link` or `yarn link`
+
+You can use [`npm link`](https://docs.npmjs.com/cli/link.html) or [`yarn link`](https://yarnpkg.com/lang/en/docs/cli/link/) to reference a package from another location on your machine in another project.
```

Comment:
What do you mean by "another project" here? Do you mean the package is in another project? Or is the Gatsby site your other project in this case?

Reply:
Yeah this does sound a little ambiguous/redundant.  I think it could be reduced to this:

```suggestion
You can use [`npm link`](https://docs.npmjs.com/cli/link.html) or [`yarn link`](https://yarnpkg.com/lang/en/docs/cli/link/) to reference a package from another location on your machine.
```

I think it's kind of saying the same thing twice, how's that suggestion?

Reply:
Looks good to me!